### DISCUSSION TITLE
cannot kekulize the scaffold after MurckoDecompose

### DISCUSSION THREAD

**Discussion by zhikaili** [rdkit/rdkit#4289] [Q&A]
Hi,

I try extracting the scaffold of a molecule (whose smiles string is 'Cc1c2cc(Br)ccc2nc-[n+]1=O'), but find that the output mol object unkekulized and hence cannot be drawn. Besides, applying AllChem.MolFromSmiles() to the output scaffold's smiles string also triggers an error about being unkekulized (but with different unkekulized atoms):

![image](https://user-images.githubusercontent.com/76140761/123746880-cdb0ef00-d8e4-11eb-82ff-2476bf41386e.png)

Btw, the output scaffold's smiles string is 'O=[n+]1[c-]nc2ccccc2c1', and can be visualized via https://pubchem.ncbi.nlm.nih.gov/ as shown below:

![image](https://user-images.githubusercontent.com/76140761/123746993-e91bfa00-d8e4-11eb-9a9e-ea45f708ff82.png)

May I ask whether the error (i.e., being unkekulized) means that the output scaffold is chemically invalid? If not, how to visualize it via rdkit?

Thank you!

**Comment by ptosco** [comment]
@zhikaili This is the same problem as in https://github.com/rdkit/rdkit/discussions/4288: implicit H counts are not updated after the bonds are cut.
Until #4290 is fixed, you can work around the problem by adjusting the H counts yourself:
```
for i in (3, 8):
    a = scaffold_mol.GetAtomWithIdx(i)
    a.SetNumExplicitHs(1)
    a.SetNoImplicit(True)
    a.UpdatePropertyCache()
Chem.SanitizeMol(scaffold_mol)
rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE
scaffold_mol
```
![image](https://user-images.githubusercontent.com/5244385/123761490-2753f200-d8c2-11eb-8ee8-2b1ee5678417.png)

**Reply by zhikaili** [reply]
Hi, thank you for your reply!
